### PR TITLE
Use (x86_64/aarch64)-linux-gnu target platform for native extensions instead of (x86_64/aarch64)-linux

### PR DESCRIPTION
**Pull request by maxirmx**
Rubygems/bundler allow  ```(x86_64/aarch64)-linux-gnu```, ```(x86_64/aarch64)-linux-musl```, ```(x86_64/aarch64)-linux``` target platform for native extensions.

When  ```(x86_64/aarch64)-linux-musl``` is not available  Ruby loads ```(x86_64/aarch64)-linux``` on Alpine Linux.
Previous versions of CI for this gem optimistically used rake-compiler defaul that created ```(x86_64/aarch64)-linux``` extension although in most cases it was not compatible with musl environment even with gcompat package installed.

So in this PR native extension target is changed to ```(x86_64/aarch64)-linux-gnu``` .  On Alpine Linux bundler/Rubygems build bantive extension from sources.

There is another issue (#105) that calls for publishing ```(x86_64/aarch64)-linux-musl``` native extension.

### CODE REVIEW

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
@maxirmx there’s a single test failure in the Windows test, seems that the tests weren’t even run. Is that something that can be fixed? Thanks!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -123,3 +130,51 @@ jobs:
...
+
+  verify:
+    name: Verify published gem on ${{ matrix.os }}
+    needs: publish
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+    steps:
```
Comment (CAMOBAP) [inline .github/workflows/release.yml:142]: It makes sense to add some sleep step here for "rubygem cooldown", to make sure last pushed gem is available through rubygems

From my own experience sometimes `gem install` right after `gem push` cannot get last pushed gem

**Review by CAMOBAP — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (maxirmx) [reply .github/workflows/release.yml:143]: Thank you, that shall help

**Comment by maxirmx** [comment]
> @maxirmx there’s a single test failure in the Windows test, seems that the tests weren’t even run. Is that something that can be fixed? Thanks!

It is the same segfault as in #139. On Windows race jsut terminates without any error message.

**Pull request merged**